consensus/spos/consensusMessageValidator_test.go

```diff
@@ -1,3 +1,455 @@
+}
+
+func TestCheckConsensusMessageValidity_InvalidPublicKeySize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	cnsMsg := &consensus.Message{ChainID: chainID, MsgType: int64(bls.MtBlockBodyAndHeader), Header: headerBytes, BlockHeaderHash: headerHash}
+	err := wrk.CheckConsensusMessageValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidPublicKeySize))
+}
+
+func TestCheckConsensusMessageValidity_InvalidSignatureSize(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	pubKey := make([]byte, PublicKeySize)
+	_, _ = rand.Read(pubKey)
+
+	cnsMsg := &consensus.Message{
+		ChainID: chainID, MsgType: int64(bls.MtBlockBodyAndHeader), Header: headerBytes, BlockHeaderHash: headerHash, PubKey: pubKey,
+	}
+	err := wrk.CheckConsensusMessageValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrInvalidSignatureSize))
+}
+
+func TestCheckConsensusMessageValidity_NodeIsNotEligible(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	pubKey := make([]byte, PublicKeySize)
+	_, _ = rand.Read(pubKey)
+	signature := make([]byte, SignatureSize)
+	_, _ = rand.Read(signature)
+
+	cnsMsg := &consensus.Message{
+		ChainID: chainID, MsgType: int64(bls.MtBlockBodyAndHeader),
+		Header: headerBytes, BlockHeaderHash: headerHash, PubKey: pubKey, Signature: signature,
+	}
+	err := wrk.CheckConsensusMessageValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrNodeIsNotInEligibleList))
+}
+
+func TestCheckConsensusMessageValidity_ErrMessageForFutureRound(t *testing.T) {
+	t.Parallel()
+
+	workerArgs := createDefaultWorkerArgs()
+	wrk, _ := spos.NewWorker(workerArgs)
+
+	headerBytes := make([]byte, 100)
+	_, _ = rand.Read(headerBytes)
+	headerHash := make([]byte, workerArgs.Hasher.Size())
+	_, _ = rand.Read(headerHash)
+	pubKey := []byte(workerArgs.ConsensusState.ConsensusGroup()[0])
+	signature := make([]byte, SignatureSize)
+	_, _ = rand.Read(signature)
+
+	cnsMsg := &consensus.Message{
+		ChainID: chainID, MsgType: int64(bls.MtBlockBodyAndHeader),
+		Header: headerBytes, BlockHeaderHash: headerHash, PubKey: pubKey, Signature: signature, RoundIndex: 10,
+	}
+	err := wrk.CheckConsensusMessageValidity(cnsMsg)
+	assert.True(t, errors.Is(err, spos.ErrMessageForFutureRound))
+}
+
+func TestCheckConsensusMessageValidity_ErrMessageForPastRound(t *testing.T) {
```

Comment:
You can add two more tests: TestCheckConsensusMessageValidity_InvalidSignature and TestCheckConsensusMessageValidity_Ok

Reply:
done.